wagon-provider-api/src/main/java/org/apache/maven/wagon/authentication/AuthenticationInfo.java

```diff
@@ -51,7 +52,300 @@
      * The absolute path to private key file
      */
     private String privateKey;
+    
+     /**
+     *
+     *
+     * The path to the trust store. If not defined, the JRE's cacert store is
+     * used.
+     *
+     *
+     */
+    private String trustStore;
+
+    /**
+     *
+     *
+     * The password to the trust store.
+     *
+     *
+     */
+    private String trustStorePassword;
```

Comment:
This field contains a password. We must convert it to `char[]` later on, so can't we store it as a `char[]`?

Reply:
The upstream model does not allow that. Given that other password fields are strings too, this is acceptable, but not ideal.

Reply:
i agree with @michael-o otherwise i would have used char[]